vm/entrypoint/src/syscalls/host.rs

```diff
@@ -41,57 +42,54 @@ pub extern "C" fn call(
 }
 
 /// Read from host storage at the given address and key.
-///
-/// The output is stored in the `key` pointer.
 #[allow(unused_variables)]
-#[no_mangle]
-pub extern "C" fn read_storage(key: *mut u32) {
+pub fn read_storage(key: &[u32; 8]) -> [u32; 8] {
   #[cfg(target_os = "zkvm")]
   unsafe {
+    let mut result = *key;
```

Comment:
Just to be clear, this does overwrite the value at `key` correct? Wouldn't it be safer and cleaner here to allocate a new memory location instead? As you do in `get_balance`

Reply:
It doesn't overwrite `key` in any place now. It creates a mutable copy of the key on the stack named `result`, which the pointer to is passed in the register (same as before). The only difference is that the code now creates a copy instead of overwriting the key, which is returned from the function "normally".